File: ui/src/app/playground/render.ts
export interface HelmRender {

  info: RenderInfo,

  chartPath? :string;
  chartName?: string;

  rawManifest?: string;

  sources?: HelmSourceFile[];
  errors?: RenderError[];

  values?: Values
  valuesFiles? : Values[];

  mergedValues?: Values;

}

export interface HelmSourceFile {
  source: string;
  manifests?: Manifest[];
}

export interface KubeConformValidation  {
  status: string
  errMsg?: string
  validationErrors?: ValidationError[]
}

export interface ValidationError {
  path: string;
  message: string;
}

export interface  RenderInfo {
  status?: string
  executionTime?: Date
}


export interface Values {
  data: string;
}

export interface GVK {
  group: string;
  version: string;
  kind: string;
}

export interface Manifest extends GVK{
  name: string;
  namespace: string;
  content: string;
  source: string;

  // Kubernetes validation
  kubeConformValidation?: KubeConformValidation

  // Yaml validation
  isYamlValid: boolean;
  yamlError: string
}


export interface  ValidationError {
  path: string;
  msg: string;
}

export interface RenderError {
  message: string;
}

File: internal/evaluation/helm/render.go
package helm

type GroupVersionKind struct {
	Group   string `json:"group"`
	Version string `json:"version"`
	Kind    string `json:"kind"`
}

type Manifest struct {
	GroupVersionKind `json:",inline"`
	Name             string `json:"name,omitempty"`
	Namespace        string `json:"manifest,omitempty"`
	Content          string `json:"content,omitempty"`

	// KubeConformValidation result
	KubeConformValidation *KubeConformValidation `json:"kubeConformValidation,omitempty"`

	IsYamlValid bool   `json:"isYamlValid"`
	YamlError   string `json:"yamlError,omitempty"`
}

type KubeConformValidation struct {
	Status           string            `json:"status,omitempty"`
	ErrMsg           string            `json:"errMsg,omitempty"`
	ValidationErrors []ValidationError `json:"validationErrors,omitempty"`
}

type ValidationError struct {
	Path string `json:"path"`
	Msg  string `json:"msg"`
}

type RenderError struct {
	Message string `json:"message,omitempty"`
}
